client/src/main/java/org/eclipse/hono/client/impl/AbstractRequestResponseClient.java

```diff
@@ -33,26 +34,51 @@
 /**
  * A Vertx-Proton based parent class for the implementation of API clients that follow the request response pattern.
  * The class is a generic that expects two classes:
- *
- * @param <C> denotes the concrete interface for the API
- * @param <R> denotes the concrete result container class of the API
- *
  * <p>
- * Both type parameter classes have their own parent and need to be subclassed.
- *
- * A subclass of this class only needs to implement some abstract helper methods (see the method descriptions) and their own
+ * Subclasses only need to implement some abstract helper methods (see the method descriptions) and their own
  * API specific methods. This allows for implementation classes that focus on the API specific code.
+ * 
+ * @param <R> The type of result this client expects the peer to return.
+ *
  */
-public abstract class AbstractRequestResponseClient<C extends RequestResponseClient, R extends RequestResponseResult<?>>
+public abstract class AbstractRequestResponseClient<R extends RequestResponseResult<?>>
         extends AbstractHonoClient implements RequestResponseClient {
 
     private static final Logger LOG = LoggerFactory.getLogger(AbstractRequestResponseClient.class);
 
-    protected final Map<String, Handler<AsyncResult<R>>> replyMap = new ConcurrentHashMap<>();
-    protected final String                               replyToAddress;
+    private final Map<String, Handler<AsyncResult<R>>> replyMap = new ConcurrentHashMap<>();
+    private final String replyToAddress;
+    private final String targetAddress;
+
+    /**
+     * Creates a request-response client.
+     * <p>
+     * The client will be ready to use after invoking {@link #createLinks(ProtonConnection)} only.
```

Comment:
For users of the class it might be helpful to give a simple check for methods like `createAndSendRequest(...)`
if they are invoked without calling `createLinks(...)`
with an appropriate error message that this call was missing.
Or what would happen if the calls was forgotten?

Reply:
very good point :+1:  I will come up with something in that direction ...